contracts/contracts/eosio.system/src/voting.cpp

```diff
@@ -84,10 +90,13 @@ namespace eosiosystem {
       auto idx = _producers.get_index<"prototalvote"_n>();
 
       std::vector< std::pair<eosio::producer_key,uint16_t> > top_producers;
-      top_producers.reserve(21);
+      top_producers.reserve(MAX_BLOCK_PRODUCERS);
 
-      for ( auto it = idx.cbegin(); it != idx.cend() && top_producers.size() < 21 && 0 < it->total_votes && it->active(); ++it ) {
-         top_producers.emplace_back( std::pair<eosio::producer_key,uint16_t>({{it->owner, it->producer_key}, it->location}) );
+      // TODO rewrite in algorithm terms
```

Comment:
we do not need to change this logic. The check producer.active() covers the case of is_satisfies_stake_requirement( producer.owner ) as it can be true only if stake above 250000

Reply:
`producer.active()` is more about having active key; producer could re-delegate\[spend?] its stake so we need to check it constantly; as well don't really want to update `active` status on stake change.